components/common-go/baseserver/testing.go

```diff
@@ -14,6 +14,18 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+// go tests does execute tests concurrently (each test a goroutine)
+// As we ran into troubles with "localhost:0" (error: "failed to start HTTP server: listen tcp 127.0.0.1:0: bind: address already in use"),
+// which might be related to a kernel bug/config issues, we're rolling our own deduplication for now
+var lastPort = 24000
+
+func getTestAddr() string {
+	// No need to lock() as we do not execute in _parallel_ but _concurrently_
```

Comment:
We do execute in parallel across packages

Reply:
That's the default for go test

Reply:
And the parallelism is further compounded by leeway doing the execution of multiple components at the same time

Reply:
Awh, you're right, we using this all over the place :grimacing: 
Will do the retry then I guess.